### PR TITLE
V3.1.2 runtime

### PR DESCRIPTION
**Pull request by jkgoodrich**
This is the code to patch the v3.1.1 frequencies with the fix for the problem identified by KC where we incorrectly handled the het non ref variants when applying the hot fix for the depletion of homozygous alternate variants. I am putting this in as a draft PR rather than changing the master repo because it is just a patch to v3.1.1. Currently this branch also doesn't have the VCF code and the HGDP + TGP subset because they are both waiting on other PRs and will need to be merged into this code when they are in master.

Code tested with the following commands:

Identify sites that need to be fixed and write out dense split MT:
```
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py --test
```
Calculate frequencies and all subset frequencies:
```
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/annotations/generate_freq_data.py --test
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/annotations/generate_freq_data.py --test --subset non_cancer
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/annotations/generate_freq_data.py --test --subset non_neuro
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/annotations/generate_freq_data.py --test --subset non_v2
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/annotations/generate_freq_data.py --test --subset non_topmed
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/annotations/generate_freq_data.py --test --subset controls_and_biobanks
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/annotations/generate_freq_data.py --test --subset tgp
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/annotations/generate_freq_data.py --test --subset hgdp
```

Patch release HT:
```
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/create_release/create_release_sites_ht.py --test --test_subsets non_v2 non_topmed non_cancer controls_and_biobanks non_neuro tgp hgdp --overwrite
```

### CODE REVIEW

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        mt = mt._filter_partitions(range(2))
+
+    logger.info(
+        "Adding annotation for whether original genotype (pre-splitting multiallelics) is het nonref..."
+    )
+    # Adding a Boolean for whether a sample had a heterozygous non-reference genotype
+    # Need to add this prior to splitting MT to make sure these genotypes
+    # are not adjusted by the homalt hotfix downstream
+    mt = mt.annotate_entries(het_non_ref=mt.LGT.is_het_non_ref())
+    # TODO: Need to fix main code to add this annotation and use it also, where to add?
```
Comment (ch-kr) [inline gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:115]: I'm not as familiar with the gnomad pipeline, but for ukbb, it would've made sense to add this annotation to the hardcalls MT

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    """
+    Filter to variants where homalt hotfix incorrectly adjusts het nonref genotype calls.
+
+    :param hl.MatrixTable mt: Raw, split MatrixTable annotated with original genotype het nonref status.
+    :param hl.Table info_ht: Info Hail Table containing AS_lowqual information.
+    :param hl.Table freq_ht: Hail Table containing 455K frequency information.
+    :return: None
+    """
+
+    logger.info("Removing low qual variants and star alleles...")
```
Comment (ch-kr) [inline gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:37]: I think you mentioned that the v3 MT doesn't have star alleles?

**Review by ch-kr — COMMENTED** [review]
a few small suggestions/questions

File: gnomad_qc/v3/create_release/create_release_sites_ht.py [hunk 3]
```diff
@@ -110,14 +108,10 @@ def add_release_annotations(freq_ht: hl.Table) -> hl.Table:
         rsid=dbsnp_ht[ht.key].rsid,
         info=info_ht[ht.key].info,
         vep=vep_ht[ht.key].vep.drop("colocated_variants"),
-        vqsr=filters_ht[ht.key].vqsr,
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v3/create_release/create_release_sites_ht.py:113]: it's actually already in the filters_ht which gets added below with `**filters_ht[ht.key]`

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py (hunk 2 above)
Comment (jkgoodrich) [reply gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:36]: Ah yes, I don't need this, I think just a copy over from your script

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    """
+    Script used to get variants impacted by homalt hotfix.
+    Most of the work was done in notebooks, and any results created in notebooks are noted with comments.
+    """
+
+    hl.init(log="/get_impacted_variants.log", default_reference="GRCh38")
+
+    logger.info("Reading sparse MT and metadata table with release only samples...")
+    mt = get_gnomad_v3_mt(
+        key_by_locus_and_alleles=True, samples_meta=True, release_only=True,
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:98]: Yes, but we are handling that subset in a different script now so this will actually only patch the full release since we actually need to redo all frequencies for the HGDP + TGP subset

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py (hunk 1 above)
Comment (jkgoodrich) [reply gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:113]: Yeah we don't have a hardcalls MT, so I think it will actually need to be added to the frequency code, but that will be another PR and I just wanted to make sure my thought was written down in the patch somewhere

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        "Writing out variants with het nonref calls that may have been incorrectly adjusted with the homalt hotfix..."
+    )
+    if args.test:
+        mt.rows().write(
+            "gs://gnomad-tmp/release_3.1.2/het_nonref_fix_sites_3.1.2_test.ht",
+            overwrite=True,
+        )
+    else:
+        mt.rows().write(
+            "gs://gnomad/release/3.1.2/ht/het_nonref_fix_sites.ht", overwrite=True
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:164]: I would need to do the filter step on 135 two times then right? I don't even think I really need this HT honestly, I really only care about the densified MT and the frequency HTs will contain this site information so I might just remove this.

File: gnomad_qc/v3/annotations/generate_freq_data.py [hunk 6]
```diff
@@ -53,50 +48,41 @@ def main(args):
...
+            "Reading dense MT containing only sites that may require frequency recalculation due to het non ref site error. "
+            "This dense MT only contains release samples and has already been split"
+        )
+        mt = hl.read_matrix_table(
+            "gs://gnomad-tmp/release_3.1.2/het_nonref_fix_sites_3.1.2_test.mt"
+            if args.test
+            else "gs://gnomad-tmp/release_3.1.2/het_nonref_fix_sites.mt"
         )
 
         if args.test:
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v3/annotations/generate_freq_data.py:81]: Yes, very true

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:164]: Actually, I think I will keep as is because this will ensure that I have a saved the HT of the sites where I recalculated frequency on the densified MT

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+
+    logger.info(
+        "Filtering to variants with at least one het nonref call and checkpointing..."
+    )
+    mt = mt.filter_rows(hl.agg.sum(mt.het_non_ref) >= 1)
+
+    logger.info(
+        "Adding new genotype entries with original homalt hotfix and het nonref-aware hotfix..."
+    )
+    mt = mt.annotate_entries(
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:54]: Haven't modified this yet, but when I make it a real PR let me know if this is what you are thining

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:98]: code for this subset has been added and frequency recalculated with the fix already put in, so marking resolved

**Comment by jkgoodrich** [comment]
Updated the PR to have resources, VCF and HGDP + 1KG subset PRs merged into it and retested using the following commands: 

```
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py --test

hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/annotations/generate_freq_data.py --test --het_nonref_patch
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/annotations/generate_freq_data.py --test --subset non_cancer --het_nonref_patch
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/annotations/generate_freq_data.py --test --subset non_neuro --het_nonref_patch
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/annotations/generate_freq_data.py --test --subset non_v2 --het_nonref_patch
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/annotations/generate_freq_data.py --test --subset non_topmed --het_nonref_patch
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/annotations/generate_freq_data.py --test --subset controls_and_biobanks --het_nonref_patch
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/annotations/generate_freq_data.py --test --subset tgp --het_nonref_patch
hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/annotations/generate_freq_data.py --test --subset hgdp --het_nonref_patch

hailctl dataproc submit jg1 PycharmProjects/gnomad_qc/gnomad_qc/v3/create_release/create_release_sites_ht.py --test --test_subsets non_v2 non_topmed non_cancer controls_and_biobanks non_neuro tgp hgdp --het_nonref_patch
```

This PR has:
- addition of `get_het_nonref_variants_to_fix.py` based on UKBB fix that Katherine created
- updates the frequency code to have a `het_nonref_patch` argument that only recalculates frequencies that need to be fixed due to the het nonref problem, without a densify if this argument is used because the MT used is already dense, and writes out the new data to files with "patch" in the name.
- modifies the sites HT release code to have a `het_nonref_patch` argument to fix the release HT based on the patch sequences
- changes to resources for the `het_nonref_patch` arguments
- changes to resource constants needed for the v3.1.2 release
- changes to use of resources because files have been deleted in favor of using the previous release HT where possible (example VEP and in silico prediction)
- a few smaller things found in the runtime of the HGDP + 1KG code and the frequency tests that needed to be fixed.

I'm struggling with the `POPS` import though. If I add `from gnomad.resources.grch38.gnomad import POPS` within `main` it works, but it doesn't currently work with the import at the top. Let me know if you have any thoughts on why?

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+        "Filtering to variants with at least one het nonref call and checkpointing..."
+    )
+    mt = mt.filter_rows(hl.agg.sum(mt._het_non_ref) >= 1)
+
+    logger.info(
+        "Adding new genotype entries with original homalt hotfix and het nonref-aware hotfix..."
+    )
+    mt = mt.annotate_entries(
+        GT_hotfix=hl.if_else(
+            mt.GT.is_het() & (mt.AF > 0.01) & (mt.AD[1] / mt.DP > 0.9),
```
Comment (ch-kr) [inline gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:57]:
```suggestion
            mt.GT.is_het() & (mt.AD[1] / mt.DP > 0.9),
```

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (ch-kr) [reply gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:55]: yeah, I think your way of filtering makes more sense than the extra `call_stats` calls. it might be easier to follow if the code runs like this:

- Filter to impacted variants using `mt = mt.filter_rows(hl.agg.any(mt.het_non_ref & ((mt.AD[1] / mt.DP) > 0.9)))`
- Adjust genotypes using code from `GT_hetnonref_fix`

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+
+    hl.init(log="/get_impacted_variants.log", default_reference="GRCh38")
+
+    logger.info("Reading sparse MT and metadata table with release only samples...")
+    mt = get_gnomad_v3_mt(
+        key_by_locus_and_alleles=True, samples_meta=True, release_only=True,
+    ).select_entries(*SPARSE_ENTRIES)
+
+    if args.test:
+        logger.info("Filtering to two partitions on chr20")
```
Comment (ch-kr) [inline gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:103]:
```suggestion
        logger.info("Filtering to two partitions on chr20...")
```
(not important, just for consistency with your other loggers)

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+
+
+def get_het_non_ref_impacted_var(
+    mt: hl.MatrixTable, info_ht: hl.Table, freq_ht: hl.Table
+) -> None:
+    """
+    Filter to variants where homalt hotfix incorrectly adjusts het nonref genotype calls.
+
+    :param hl.MatrixTable mt: Raw, split MatrixTable annotated with original genotype het nonref status.
+    :param hl.Table info_ht: Info Hail Table containing AS_lowqual information.
```
Comment (ch-kr) [inline gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:39]: this isn't getting used in this function, is it? we should probably remove AS_lowqual variants after densifying anyway, right?

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+    logger.info("Densifying MT to het nonref sites only...")
+    # NOTE: densify_sites operates on locus only (doesn't check alleles)
+    # https://github.com/broadinstitute/gnomad_methods/blob/master/gnomad/utils/sparse_mt.py#L102
+    # Thus, it doesn't matter that `sites_ht` has already been split
+    # NOTE: set semi_join_rows to False here because the sites_HT is small
+    mt = densify_sites(
+        mt, sites_ht, last_END_position.versions["3.1"].ht(), semi_join_rows=False,
+    )
+    mt = mt.filter_rows(
+        (hl.len(mt.alleles) > 1) & (hl.is_defined(sites_ht[mt.row_key]))
```
Comment (ch-kr) [inline gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:147]: do you want to filter lowqual variants here?

File: gnomad_qc/v3/annotations/generate_freq_data.py [hunk 12]
```diff
@@ -142,9 +157,10 @@ def main(args):
             adj=get_adj_expr(mt.GT, mt.GQ, mt.DP, mt.AD),
         )
 
-        logger.info("Densify-ing...")
-        mt = hl.experimental.densify(mt)
-        mt = mt.filter_rows(hl.len(mt.alleles) > 1)
+        if not args.het_nonref_patch:
+            logger.info("Densify-ing...")
```
Comment (ch-kr) [inline gnomad_qc/v3/annotations/generate_freq_data.py:161]:
```suggestion
            logger.info("Densifying...")
```
(there's no hyphen in the other script)

File: gnomad_qc/v3/annotations/generate_freq_data.py [hunk 13]
```diff
@@ -50,6 +50,8 @@
 
 
 def main(args):
+    from gnomad.resources.grch38.gnomad import POPS
```
Comment (ch-kr) [inline gnomad_qc/v3/annotations/generate_freq_data.py:53]: I just figured it out, the POPS import is broken because of lines 194-195:
```
if n_subsets_use_subpops:
    POPS = POPS_STORED_AS_SUBPOPS
```
I think python is angry or confused with the reassignment of POPS. might be a hacky fix, but you could do something like
```
pops = POPS 
if n_subsets_use_subpops:
    pops = POPS_STORED_AS_SUBPOPS
```

File: gnomad_qc/v3/create_release/create_hgdp_tgp_subset.py [hunk 14]
```diff
@@ -466,6 +466,24 @@ def adjust_subset_alleles(mt: hl.MatrixTable) -> hl.MatrixTable:
     :param mt: MatrixTable to subset locus alleles
     :return: MatrixTable with alleles adjusted to only those with a sample containing a non reference allele
     """
+
+    def split_shuffle(mt: hl.MatrixTable) -> hl.MatrixTable:
+        """
+        Split rows that do and do not have the same new and old locus (which will be most) prior to the row rekey.
```
Comment (ch-kr) [inline gnomad_qc/v3/create_release/create_hgdp_tgp_subset.py:472]:
```suggestion
        Split rows that do and do not have the same new and old locus prior to the row rekey.
        
        Most rows will have the same new and old locus annotations.
```
optional, this seemed slightly easier to read to me

File: gnomad_qc/v3/create_release/create_hgdp_tgp_subset.py [hunk 15]
```diff
@@ -466,6 +466,24 @@ def adjust_subset_alleles(mt: hl.MatrixTable) -> hl.MatrixTable:
...
+
+        This is needed to force the shuffle to only happen on the small number of rows that have a different new and
```
Comment (ch-kr) [inline gnomad_qc/v3/create_release/create_hgdp_tgp_subset.py:474]:
```suggestion
        Re-keying rows triggers a shuffle in hail.
        This function is needed to force the shuffle to only happen on the small number of rows that have a different new and
```

File: gnomad_qc/v3/create_release/create_hgdp_tgp_subset.py [hunk 16]
```diff
@@ -466,6 +466,24 @@ def adjust_subset_alleles(mt: hl.MatrixTable) -> hl.MatrixTable:
...
+        old locus. Fixing a problem discussed in more detail here: https://discuss.hail.is/t/preventing-a-shuffle-error/2261/6
+
+        :param mt: MatrixTable to rekey
+        :return: Rekeyed MatrixTable
+        """
+        mt1 = mt.filter_rows(mt.locus == mt.new_locus)
```
Comment (ch-kr) [inline gnomad_qc/v3/create_release/create_hgdp_tgp_subset.py:480]:
```suggestion
        # Filter to rows that have the same old and new locus annotations
        # These rows do not trigger a shuffle when re-keyed
        mt1 = mt.filter_rows(mt.locus == mt.new_locus)
```

File: gnomad_qc/v3/create_release/create_hgdp_tgp_subset.py [hunk 17]
```diff
@@ -466,6 +466,24 @@ def adjust_subset_alleles(mt: hl.MatrixTable) -> hl.MatrixTable:
...
+        mt1 = mt1.key_rows_by(locus=mt1.new_locus, alleles=mt1.new_alleles)
+        mt2 = mt.filter_rows(mt.locus != mt.new_locus)
```
Comment (ch-kr) [inline gnomad_qc/v3/create_release/create_hgdp_tgp_subset.py:482]:
```suggestion
        # Filter to row with different old and new locus annotations
        # These rows will trigger a shuffle when re-keyed
        mt2 = mt.filter_rows(mt.locus != mt.new_locus)
```

File: gnomad_qc/v3/create_release/create_release_sites_ht.py [hunk 18]
```diff
@@ -193,49 +220,43 @@ def pre_process_subset_freq(
...
-            global_freq_ht = (
-                hl.read_table(get_freq().path)
-                .select("freq")
-                .select_globals("freq_meta")
+                "Loading global frequency data for testing: %s", global_freq_path
             )
+            global_freq_ht = hl.read_table(global_freq_path)
             global_freq_ht = hl.filter_intervals(
-                global_freq_ht, [hl.parse_locus_interval("chr20:1-1000000")]
+                global_freq_ht, [hl.parse_locus_interval("chr1:1-1000000")]
```
Comment (ch-kr) [inline gnomad_qc/v3/create_release/create_release_sites_ht.py:254]: basic question, why the change from chr20-chr1? was it too slow to filter the freq ht to chr20 for testing?

**Review by ch-kr — CHANGES_REQUESTED** [review]
a few small suggestions and questions

File: gnomad_qc/v3/create_release/create_release_sites_ht.py (hunk 18 above)
Comment (jkgoodrich) [reply gnomad_qc/v3/create_release/create_release_sites_ht.py:254]: No, I just used the first 2 partitions in the frequency test code, so this will allow me to still test that those frequencies are being added, otherwise they are all missing. I also didn't fully understand the original use of chr20.

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py (hunk 9 above)
Comment (jkgoodrich) [reply gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:103]: I will also change this to chr1 as I did for the other

File: gnomad_qc/v3/create_release/hgdp_tgp_constants.py [hunk 19]
```diff
@@ -418,7 +418,7 @@
                     "https://www.internationalgenome.org/category/population."
                 )
             },
-            "geographic_region": {
+            "genetic_region": {
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v3/create_release/hgdp_tgp_constants.py:421]: Request from Alicia Martin. They prefer to call it the genetic region not geographic

File: gnomad_qc/v3/annotations/generate_freq_data.py (hunk 13 above)
Comment (jkgoodrich) [reply gnomad_qc/v3/annotations/generate_freq_data.py:53]: Thank you!

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py (hunk 10 above)
Comment (jkgoodrich) [reply gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:39]: Nope, no longer used. AS_lowqual are already removed from the freq_ht, so I don't think I need to remove them anywhere

Comment (jkgoodrich) [reply gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:39]: So I don't even need the info HT actually

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py (hunk 11 above)
Comment (jkgoodrich) [reply gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:147]: I thought I needed to, but actually don't since they are not in the freq_ht

File: gnomad_qc/v3/resources/release.py [hunk 20]
```diff
@@ -44,54 +44,68 @@ def release_ht_path(
...
     :rtype: str
     """
     version_prefix = "r" if release_version.startswith("3.0") else "v"
     if public:
+        if het_nonref_patch:
+            DataException("The patch HT will not be made public")
         return f"gs://gnomad-public-requester-pays/release/{release_version}/ht/{data_type}/gnomad.{data_type}.{version_prefix}{release_version}.sites.ht"
     else:
-        return f"gs://gnomad/release/{release_version}/ht/{data_type}/gnomad.{data_type}.{version_prefix}{release_version}.sites.ht"
+        return f"gs://gnomad/release/{release_version}/ht/{data_type}/gnomad.{data_type}.{version_prefix}{release_version}{'.public' if het_nonref_patch else ''}.sites.ht"
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v3/resources/release.py:65]: Thanks for the catch, it is supposed to say `patch`, I had public on my mind

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py [hunk 21]
```diff
@@ -47,42 +44,7 @@ def get_het_non_ref_impacted_var(
...
-    mt = mt.transmute_rows(hom_count_hotfix=mt.gt_stats.homozygote_count[1])
-
-    mt = mt.annotate_rows(gt_stats=hl.agg.call_stats(mt.GT_hetnonref_fix, mt.alleles))
-    mt = mt.transmute_rows(hom_count_hetnonref_fix=mt.gt_stats.homozygote_count[1])
-
-    logger.info(
-        "Filtering to rows where homalt hotfix erroneously adjusts het nonref calls..."
-    )
-    mt = mt.filter_rows(mt.hom_count_hotfix != mt.hom_count_hetnonref_fix)
+    mt = mt.filter_rows(hl.agg.any(mt.het_non_ref & ((mt.AD[1] / mt.DP) > 0.9)))
```
Comment (ch-kr) [inline gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:47]:
```suggestion
    return mt.filter_rows(hl.agg.any(mt.het_non_ref & ((mt.AD[1] / mt.DP) > 0.9)))
```

**Review by ch-kr — APPROVED** [review]
one more tiny suggestion but LGTM!

File: gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py (hunk 21 above)
Comment (ch-kr) [reply gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py:47]: (and remove the `return mt` below)
